node/src/components/event_stream_server/tests.rs

```diff
@@ -111,6 +126,8 @@ impl ServerSyncBehavior {
     async fn sleep_if_required(&self) {
         if self.has_delay_between_events {
             time::sleep(DELAY_BETWEEN_EVENTS).await;
+        } else {
+            task::yield_now().await;
```

Comment:
I'm not sure this is necessary, since we're calling an `async` function, thus giving the runtime a chance to intercept anyway? :)

Reply:
Yes, it shouldn't be strictly necessary.  However, the local tests seemed to benefit from this in the case of the lagging node.  My thinking is that this helps interleave the time slices more evenly between the client task and the server task.  We need the server task to actually progress (without a sleep) in order for the server to fill the buffers and kick the client off.  

However, it the server gets too big a time slice, it gets a chance to enqueue a _lot_ more events to the stream.  The only thing stopping the server doing that in its time slice is for the clients to detect the test has ended, allowing their futures to join and the call to stop the running server can happen.

So, it benefits us to give the clients control as frequently as possible.